### PR TITLE
Implement framebuffer fetch for Metal backend

### PR DESCRIPTION
**Pull request by bejado**
This saves about .5ms on iOS.

### CODE REVIEW

File: filament/backend/src/metal/MetalDriver.mm [hunk 1]
```diff
@@ -917,8 +921,22 @@
...
+    }
+    if (any(buffers & TargetBufferFlags::COLOR2)) {
+        srcAttachIndex = 2;
+    }
+    if (any(buffers & TargetBufferFlags::COLOR3)) {
+        srcAttachIndex = 3;
+    }
+    MetalRenderTarget::Attachment srcColorAttachment = srcTarget->getColorAttachment(srcAttachIndex);
+
+    // We always blit to the COLOR0 attachment.
```
Comment (pixelflinger) [inline filament/backend/src/metal/MetalDriver.mm:939]: This seems unrelated?

File: filament/backend/src/metal/MetalDriver.mm [hunk 2]
```diff
@@ -934,9 +952,12 @@
             srcTarget->height - (NSUInteger) srcRect.bottom - srcRect.height,
             srcRect.width, srcRect.height);
 
+    const NSInteger dstHeight =
+            dstTarget->isDefaultRenderTarget() ?
+            mContext->currentSurface->getSurfaceHeight() : dstTarget->height;
     MTLRegion dstRegion = MTLRegionMake2D(
             (NSUInteger) dstRect.left,
-            dstTarget->height - (NSUInteger) dstRect.bottom - dstRect.height,
+            dstHeight - (NSUInteger) dstRect.bottom - dstRect.height,
```
Comment (pixelflinger) [inline filament/backend/src/metal/MetalDriver.mm:960]: This also seems unrelated?

Comment (bejado) [reply filament/backend/src/metal/MetalDriver.mm:960]: Technically unrelated, but this bug was "exposed" by framebuffer fetch.

File: filament/backend/src/metal/MetalDriver.mm (hunk 1 above)
Comment (bejado) [reply filament/backend/src/metal/MetalDriver.mm:939]: I originally added this because I thought we needed it (we're bliting from COLOR1), though it looks like you create a new RT for the blit so it isn't strictly needed. But the way it was written before was incorrect, so probably best to leave it.
